modules/dxp/apps/segments/segments-asah-connector-test/src/testIntegration/java/com/liferay/segments/asah/connector/internal/model/listener/test/LayoutModelListenerTest.java

```diff
@@ -0,0 +1,190 @@
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getCompanyId(), _group.getGroupId(),
+			TestPropsValues.getUserId());
+	}
+
+	@Test
+	public void testOnBeforeUpdate() throws Exception {
+		Layout layout1 = LayoutTestUtil.addTypeContentLayout(_group);
+
+		long defaultSegmentsExperienceId =
+			_segmentsExperienceLocalService.fetchDefaultSegmentsExperienceId(
+				layout1.getPlid());
+
+		SegmentsExperiment segmentsExperiment =
+			_segmentsExperimentLocalService.addSegmentsExperiment(
+				defaultSegmentsExperienceId,
+				_portal.getClassNameId(Layout.class), layout1.getPlid(),
+				"AB test", "A/B test description",
+				SegmentsExperimentConstants.Goal.BOUNCE_RATE.getLabel(),
+				StringPool.BLANK, _serviceContext);
+
+		Object asahSegmentsExperimentProcessor =
+			ReflectionTestUtil.getFieldValue(
+				_modelListener, "_asahSegmentsExperimentProcessor");
+
+		Object asahFaroBackendClient = ReflectionTestUtil.getFieldValue(
+			asahSegmentsExperimentProcessor, "_asahFaroBackendClient");
+
+		ReflectionTestUtil.setFieldValue(
+			asahFaroBackendClient, "_http",
+			MockHttpUtil.geHttp(
+				Collections.singletonMap(
+					"/api/1.0/experiments/" +
+						segmentsExperiment.getSegmentsExperienceId(),
+					() -> "{}")));
+
+		SegmentsExperience segmentsExperience =
+			_segmentsExperienceLocalService.addSegmentsExperience(
+				TestPropsValues.getUserId(), _group.getGroupId(), 0,
+				_portal.getClassNameId(Layout.class), layout1.getPlid(),
+				Collections.singletonMap(
+					LocaleUtil.getSiteDefault(), "Variant 1"),
+				false, new UnicodeProperties(true), _serviceContext);
+
+		_segmentsExperimentRelService.addSegmentsExperimentRel(
+			segmentsExperiment.getSegmentsExperimentId(),
+			segmentsExperience.getSegmentsExperienceId(), _serviceContext);
+
+		try (CompanyConfigurationTemporarySwapper
+				companyConfigurationTemporarySwapper =
+					new CompanyConfigurationTemporarySwapper(
+						TestPropsValues.getCompanyId(),
+						AnalyticsConfiguration.class.getName(),
+						HashMapDictionaryBuilder.<String, Object>put(
+							"liferayAnalyticsDataSourceId",
+							RandomTestUtil.randomLong()
+						).put(
+							"liferayAnalyticsEnableAllGroupIds", true
+						).put(
+							"liferayAnalyticsFaroBackendSecuritySignature",
+							RandomTestUtil.randomString()
+						).put(
+							"liferayAnalyticsFaroBackendURL",
+							RandomTestUtil.randomString()
+						).build(),
+						SettingsFactoryUtil.getSettingsFactory())) {
+
+			String title = "Layout title modified";
+
+			layout1.setTitleMap(
+				Collections.singletonMap(LocaleUtil.getSiteDefault(), title));
+
+			_layoutLocalService.updateLayout(layout1);
+
+			Layout layout2 = _layoutLocalService.getLayout(layout1.getPlid());
```

Comment:
looks like layout2 is not layout 2, but the layout1 after update, right?

Then we should follow the pattern initialLayout, actualLayout